phpBB/includes/functions_download.php

```diff
@@ -41,15 +43,12 @@ function send_avatar_to_browser($file, $browser)
 	{
 		$image_dir = '';
 	}
-	$file_path = $phpbb_root_path . $image_dir . '/' . $prefix . $file;
+	$file_path = $image_dir . '/' . $prefix . $file;
 
-	if ((@file_exists($file_path) && @is_readable($file_path)) && !headers_sent())
+	if ($storage->exists($file_path) && !headers_sent())
 	{
 		header('Cache-Control: public');
 
-		$image_data = @getimagesize($file_path);
-		header('Content-Type: ' . image_type_to_mime_type($image_data[2]));
```

Comment:
Use fast-image-size for this is a little complicated, i seeing that the api of some storage services allow to retrieve the mimetype and file size.
I can create new methods in storage (something like get_size and get_mimetype) that throws an exception if those methods arent available, in that case i dont send the headers.
What do you think?
About streams, yes, is something i have to do to not load the entire files in memory when they are downloaded or something